get_azure_unique_id

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+#
+# get_azure_unique_id: prints a unique id for an Azure VM suitable for use with SignalFx.
+# exits with 0 if the unique id was generated succesfully
+# exits with 1 if the curl for data succeeded but the parsing of the information failed.
+# exits with 28 if there was a timeout talking to Azure to get information
+# all other exit codes come from curl
+#
+
+jsonval () {
+    temp=`echo $json | sed 's/\\\\\//\//g' | sed 's/[{}]//g' | awk -v k="text" '{n=split($0,a,","); for (i=1; i<=n; i++) print a[i]}' | sed 's/\"\:\
+"/\|/g' | sed 's/[\,]/ /g' | sed 's/\"//g' | grep -w $1`
+    echo ${temp##*|}
+}
+
+json=`curl -H Metadata:true "http://169.254.169.254/metadata/instance?api-version=2017-08-01"`
+status=$?
+if [ $status -ne 0 ]; then
+        exit $status
+fi
+
+subId=$(jsonval "subscriptionId")
+rgname=$(jsonval "resourceGroupName")
+name=$(jsonval "name")
+
+if [ ! -z $subId -a ! -z $rgname -a ! -z $name ]; then
+  temp=$subId'/'$rgname'/microsoft.compute/virtualmachines/'$name
+  final= echo $temp | awk '{print tolower($0)}'
```

Comment:
I'm not familiar with this syntax where there is a space after the = and a command following.

Reply:
I noticed that too. Without that I the method errors out with: `./get_azure_unique_id: line 29: 1b582ebf-2f11-49d7-bcc5-c468717d50e2/sfdemo/microsoft.compute/virtualmachines/storage-feeder: No such file or directory`. I think it's something to do with the fact that the string assigned to `final` is the output of a command. Not sure though.